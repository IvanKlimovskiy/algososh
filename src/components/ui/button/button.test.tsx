import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Тестирование компонента Button', () => {
  test('Button правильно рендерится с текстом', () => {
    const buttonText = 'Нажми меня';
    const tree = renderer.create(<Button text={buttonText} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Button правильно рендерится без текста', () => {
    const buttonText = '';
    const tree = renderer.create(<Button text={buttonText} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Заблокированный Button правильно рендерится', () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Заблокированный Button правильно рендерится', () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Проверка корректности вызова колбэка', () => {
    const onClick = jest.fn();
    const buttonText = 'Нажми меня';
    render(<Button text={buttonText} onClick={onClick} />);
    const link = screen.getByText('Нажми меня');
    fireEvent.click(link);
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
