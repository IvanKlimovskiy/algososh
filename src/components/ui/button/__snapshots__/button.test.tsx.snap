// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Button Button правильно рендерится без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Button правильно рендерится с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми меня
  </p>
</button>
`;

exports[`Тестирование компонента Button Заблокированный Button правильно рендерится 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Заблокированный Button правильно рендерится 2`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
